src/rules/swarm.ts

```diff
@@ -0,0 +1,141 @@
+import { YAMLRule } from "../lint";
+
+export const swarmSecretNameValue: YAMLRule = {
+  name: "prop-swarm-sectet-name-value",
+  type: "error",
+  message: "Swarm secrets require both a `name` and a `value` to function.",
+  test: {
+    AnyOf: {
+      path: "swarm",
+      pred: {
+        AnyOf: {
+          path: "secrets",
+          pred: {
+            Or: {
+              preds: [
+                { IsEmpty: { path: "name"} },
+                { IsEmpty: { path: "value"} },
+                { Eq: { path: "name", value: ""} },
+                { Eq: { path: "value", value: ""} },
+              ],
+            },
+          },
+        },
+      },
+    },
+  },
+  examples: {
+    wrong: [
+      {
+        description: "A swarm secret must contain a `name` and a `value`",
+        yaml: `
+---
+replicated_api_version: "2.7.0"
+swarm:
+- secrets:
+  - name: foo
+        `,
+      },
+      {
+        description: "A swarm secret must contain a `name` and a `value`",
+        yaml: `
+---
+replicated_api_version: "2.7.0"
+swarm:
+- secrets:
```

Comment:
I don't think `swarm` itself should be an array of items: https://github.com/replicatedhq/libyaml/blob/master/swarm.go#L26, I think you'd want something like 

```
swarm:
  secrets:
    - name: ...
      value: ....

```

Reply:
Right, I'll fix that